src/Tools/dotnet-monitor/CollectionRules/CollectionRulePipeline.cs

```diff
@@ -136,38 +142,25 @@ protected override async Task OnRun(CancellationToken token)
                         }
                     }
 
-                    DateTime currentTimestamp = _context.Clock.UtcNow.UtcDateTime;
+                    DateTime currentTimestamp = Context.Clock.UtcNow.UtcDateTime;
 
-                    // If rule has an action count window, Remove all execution timestamps that fall outside the window.
-                    if (actionCountWindowDuration.HasValue)
+                    if (_stateHolder.CanExecuteActions(currentTimestamp))
                     {
-                        DateTime windowStartTimestamp = currentTimestamp - actionCountWindowDuration.Value;
-                        while (executionTimestamps.Count > 0)
+                        lock (_stateHolder.ExecutionTimestamps)
```

Comment:
We lock around ExecutionTimeStamps here but around _lock when copying. Can we just use a single lock?